src/main/kotlin/me/ezra_home/retail_software_solution/rest/endpoints/sysuser/FruitEndpoint.kt

```diff
@@ -0,0 +1,78 @@
+package me.ezra_home.retail_software_solution.rest
+
+import me.ezra_home.retail_software_solution.businesses.sysuser.service.FruitService
+import me.ezra_home.retail_software_solution.model.entity.FruitEntity
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.util.UUID
+
+@RestController
+@RequestMapping("/api/fruits")
+class FruitController(
+    private val fruitService: FruitService
+) {
+
+    /**
+     * Get all fruits created by a specific user.
+     * @param userId UUID of the user.
+     * @return A list of FruitEntity.
+     */
+    @GetMapping("/user/{userId}")
+    fun getFruitsByUser(@PathVariable userId: UUID): ResponseEntity<List<FruitEntity>> {
+        val fruits = fruitService.getFruitsByUserId(userId)
+        return ResponseEntity.ok(fruits)
+    }
+
+    /**
+     * Add a new fruit.
+     * @param fruit Request body containing the fruit details.
+     * @return The saved FruitEntity.
+     */
+    @PostMapping
+    fun addFruit(@RequestBody fruit: FruitEntity): ResponseEntity<FruitEntity> {
```

Comment:
I dont think we need to wrap our responses on `ResponseEntity`. It adds no value